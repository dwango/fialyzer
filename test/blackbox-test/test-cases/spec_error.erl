-module(spec_error).

-export([foo/1, int1/0, int2/0, int3/0, int4/0,
         list1/0, list2/0, list3/0, list4/0, list5/0, string1/0, string2/0,
         map1/0, map2/0,
         bar/2]).


-spec foo(number()) -> boolean().
foo(111) -> ok.

-spec int1() -> non_neg_integer().
int1() -> ok.

-spec int2() -> pos_integer().
int2() -> ok.

-spec int3() -> neg_integer().
int3() -> ok.

-spec int4() -> integer().
int4() -> ok.

-spec list1() -> [] | nil().
list1() -> ok.

-spec list2() -> list().
list2() -> ok.

-spec list3() -> list(atom()).
list3() -> ok.

-spec list4() -> nonempty_list(atom()).
list4() -> ok.

-spec list5() -> maybe_improper_list(atom(), number()).
list5() -> ok.

-spec string1() -> string().
string1() -> ok.

-spec string2() -> nonempty_string().
string2() -> ok.

-spec map1() -> map().
map1() -> ok.

-spec map2() -> #{hoge := atom(), number() := atom()}.
map2() -> ok.

-spec bar(boolean(), number()) -> ok.
bar(M, N) ->
    _ = M + N,
    ok.
