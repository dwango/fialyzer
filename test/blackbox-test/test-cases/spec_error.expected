test-cases/spec_error.erl:10: foo: Type spec unmatch;
  success type: fun((111) -> 'ok')
  type spec   : fun((number()) -> 'true' | 'false')
test-cases/spec_error.erl:13: int1: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> number())
test-cases/spec_error.erl:16: int2: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> number())
test-cases/spec_error.erl:19: int3: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> number())
test-cases/spec_error.erl:22: int4: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> number())
test-cases/spec_error.erl:25: list1: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> [none()])
test-cases/spec_error.erl:28: list2: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> [any()])
test-cases/spec_error.erl:31: list3: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> [atom()])
test-cases/spec_error.erl:34: list4: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> [atom()])
test-cases/spec_error.erl:37: list5: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> [atom()])
test-cases/spec_error.erl:40: string1: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> [number()])
test-cases/spec_error.erl:43: string2: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> [number()])
test-cases/spec_error.erl:46: map1: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> map())
test-cases/spec_error.erl:49: map2: Type spec unmatch;
  success type: fun(() -> 'ok')
  type spec   : fun(() -> map())
test-cases/spec_error.erl:53: bar: Type spec unmatch;
  success type: fun((number(), number()) -> 'ok')
  type spec   : fun(('true' | 'false', number()) -> 'ok')
test-cases/spec_error.erl:63: user_error: Type spec unmatch;
  success type: fun(({'typ1', 'typ2'}) -> 'ok')
  type spec   : fun(({'typ2', 'typ1'}) -> 'ok')
